### PR TITLE
FE-2171 RadioGroup now uses fieldset and legend to support Web Accessi…

### PR DESCRIPTION
**Pull request by damianOfKyan**
RadioGroup will now use Legend instead of Label for better Web Accessibility support.

### CODE REVIEW

File: src/__experimental__/components/radio-button/radio-button-group.component.js [hunk 1]
```diff
@@ -60,7 +54,7 @@ RadioButtonGroup.propTypes = {
   /** Specifies the name prop to be applied to each button in the group */
   groupName: PropTypes.string.isRequired,
   /** The content for the RadioGroup Label */
```
Comment (jamime) [inline src/__experimental__/components/radio-button/radio-button-group.component.js]:
```suggestion
   /** The content for the RadioGroup Legend */
```

File: src/__experimental__/components/radio-button/radio-button-group.spec.js [hunk 2]
```diff
@@ -18,7 +16,7 @@ function render(renderer = TestRenderer.create) {
   return renderer(
     <RadioButtonGroup
       groupName={ groupName }
-      label='Test RadioButtonGroup Label'
+      legend='Test RadioButtonGroup Label'
```
Comment (jamime) [inline src/__experimental__/components/radio-button/radio-button-group.spec.js]:
```suggestion
      legend='Test RadioButtonGroup Legend'
```

File: src/__experimental__/components/radio-button/radio-button-group.spec.js [hunk 3]
```diff
@@ -79,7 +77,7 @@ describe('RadioButtonGroup', () => {
           const radioGroup = shallow(
             <RadioButtonGroup
               groupName={ groupName }
-              label='Test RadioButtonGroup Label'
+              legend='Test RadioButtonGroup Label'
```
Comment (jamime) [inline src/__experimental__/components/radio-button/radio-button-group.spec.js]:
```suggestion
              legend='Test RadioButtonGroup Legend'
```

File: src/__experimental__/components/radio-button/radio-button.stories.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ function makeStory(name, themeSelector) {
     return (
       <RadioButtonGroup
         groupName='frequency'
-        label={ text('groupLabel', 'Please select a frequency from the options below') }
+        legend={ text('groupLabel', 'Please select a frequency from the options below') }
```
Comment (jamime) [inline src/__experimental__/components/radio-button/radio-button.stories.js]:
```suggestion
        legend={ text('legend', 'Please select a frequency from the options below') }
```

Comment (damianOfKyan) [reply src/__experimental__/components/radio-button/radio-button.stories.js]: resolved in [https://github.com/Sage/carbon/pull/2395/commits/8ad77ae94edd822af3a2c8bee9159daa445c16b8](8ad77ae)

File: src/__experimental__/components/radio-button/radio-button-group.spec.js (hunk 3 above)
Comment (damianOfKyan) [reply src/__experimental__/components/radio-button/radio-button-group.spec.js]: resolved in [https://github.com/Sage/carbon/pull/2395/commits/8ad77ae94edd822af3a2c8bee9159daa445c16b8](8ad77ae)

File: src/__experimental__/components/radio-button/radio-button-group.spec.js (hunk 2 above)
Comment (damianOfKyan) [reply src/__experimental__/components/radio-button/radio-button-group.spec.js]: resolved in [https://github.com/Sage/carbon/pull/2395/commits/8ad77ae94edd822af3a2c8bee9159daa445c16b8](8ad77ae)

File: src/__experimental__/components/radio-button/radio-button-group.component.js [hunk 5]
```diff
@@ -38,12 +38,12 @@ const RadioButtonGroup = (props) => {
 
   return (
     <RadioFieldsetStyle legend={ legend }>
-      <StyledRadioButtonGroup
+      <div
         role='radiogroup'
```
Comment (jamime) [inline src/__experimental__/components/radio-button/radio-button-group.component.js]: `role='radiogroup'` is not a valid HTML attribute. Let's get rid of the `div` all together and move `tagComponent` to `RadioFieldsetStyle`. Thanks @damianOfKyan!

Comment (damianOfKyan) [reply src/__experimental__/components/radio-button/radio-button-group.component.js]: resolved in 59ca991

**Pull request merged**